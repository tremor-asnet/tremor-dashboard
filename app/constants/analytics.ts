/* Constants for Analytics Page */
export const SALE_STATISTICAL_TYPE = {
  REVENUE: "Revenue",
};
