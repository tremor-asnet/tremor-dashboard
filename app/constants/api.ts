import { ROUTER_API_URL } from "./routes";

export const API_ROUTES = {
  USERS: `${ROUTER_API_URL}/users`,
};
