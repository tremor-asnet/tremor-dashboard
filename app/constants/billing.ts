export const enum AGGREGATION_TYPE {
  SALARY = "Salary",
  PAYPAL = "Paypal",
}

export const enum AGGREGATION_DESCRIPTION {
  SALARY = "Belong Interactive",
  PAYPAL = "Freelance Payment",
}
