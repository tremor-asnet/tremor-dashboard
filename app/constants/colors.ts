type ColorChannelsChartType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
};

export const CHANNELS_CHART_COLOR: ColorChannelsChartType = {
  1: "blue-500",
  2: "red-500",
  3: "primary",
  4: "gray-500",
};

export const TRANSACTION_COLOR = {
  INCREASE: "few",
  DECREASE: "attention",
  PENDING: "secondary",
};
