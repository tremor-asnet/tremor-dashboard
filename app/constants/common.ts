export const EXCEPT_KEYS = {
  POSITIVE_INTEGER: ["e", "E", "+", "-", "."],
  POSITIVE_DOUBLE: ["e", "E", "+", "-"],
};

export const DIRECTION = {
  ASC: "asc",
  DESC: "desc",
};
