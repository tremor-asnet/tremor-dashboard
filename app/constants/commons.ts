export const ITEM_ACTION_PROJECT = [
  {
    key: "act",
    label: "Action",
  },
  {
    key: "anotherAct",
    label: "Another action",
  },
  {
    key: "somethingElse",
    label: "Something else here",
  },
];

export const ITEM_ACTION_SALES_DATE = [
  {
    key: "last7Days",
    label: "Last 7 days",
  },
  {
    key: "lastWeek",
    label: "Last week",
  },
  {
    key: "last30Days",
    label: "Last 30 days",
  },
];

export const URL_AVATAR_PROJECT = [
  {
    avatar: "/images/avatar/avatar-xs.webp",
  },
  {
    avatar: "/images/avatar/avatar-xs.webp",
  },
  {
    avatar: "/images/avatar/avatar-xs.webp",
  },
];

// Social link
export const SOCIAL_LINK = {
  facebook: "https://www.facebook.com/",
  twitter: "https://twitter.com/",
  instagram: "https://instagram.com",
};

export const REMEMBER_ME = "remember-me";

export const listOption: ListOptionProps[] = [
  { option: "Paid" },
  { option: "Refunded" },
  { option: "Canceled" },
];
