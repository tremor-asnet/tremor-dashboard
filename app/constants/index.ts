export * from "./routes";
export * from "./regex";
export * from "./messages";
export * from "./profile";
export * from "./commons";
export * from "./flagSrc";
export * from "./images";
export * from "./sections";
export * from "./images";
export * from "./products";
export * from "./colors";
export * from "./units";
export * from "./sales";
export * from "./commons";
export * from "./status";
