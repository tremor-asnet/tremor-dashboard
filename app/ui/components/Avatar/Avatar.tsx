"use client";

import { memo } from "react";

// Constants
import { AVATAR_SRC } from "@/constants";

// Components
import { CustomImage } from "@/ui/components";

// Define the props for the Input component
interface AvatarProps {
  src: string;
  width: number;
  height: number;
  alt: string;
  className?: string;
  sizes?: string;
  priority?: boolean;
}

/**
 * Primary UI component for Avatar component
 */
const Avatar = ({
  src,
  width,
  height,
  alt,
  className = "shadow-md",
  sizes = AVATAR_SRC.md,
  priority = true,
}: AvatarProps): JSX.Element => {
  return (
    <div className={`inline-flex rounded-full overflow-hidden ${className}`}>
      {src ? (
        <CustomImage
          src={src}
          width={width}
          height={height}
          alt={alt}
          sizes={sizes}
          priority={priority}
        />
      ) : (
        <p
          className={`w-6 flex justify-center items-center text-white text-xs bg-gray-500 rounded-full`}>
          {alt.substring(0, 1)}
        </p>
      )}
    </div>
  );
};

export default memo(Avatar);
