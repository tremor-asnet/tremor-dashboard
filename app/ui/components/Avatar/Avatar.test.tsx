import { render } from "@testing-library/react";

// Components
import Avatar from "./Avatar";

// Constants
import { AVATAR_SRC } from "@/constants";

describe("Testing avatar component", () => {
  const propsDefault = {
    alt: "Avatar md",
    className: "shadow-md",
    height: 48,
    priority: true,
    src: AVATAR_SRC.md,
    width: 48,
  };

  const propsEmptyOptions = {
    alt: "",
    height: NaN,
    width: NaN,
    src: "",
  };

  it("Should match snapshot", () => {
    const component = render(<Avatar {...propsDefault} />);
    expect(component).toMatchSnapshot();
  });

  it("Should test empty props", () => {
    const { container } = render(<Avatar {...propsEmptyOptions} />);
    const elementImg = container.querySelector("img");
    expect(elementImg).toBeFalsy();
    const elementparagraph = container.querySelector("p");
    expect(elementparagraph).toBeTruthy();
    expect(elementparagraph.textContent).toBe("");
  });
});
