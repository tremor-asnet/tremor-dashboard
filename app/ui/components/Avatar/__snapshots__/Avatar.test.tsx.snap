// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing avatar component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="inline-flex rounded-full overflow-hidden shadow-md"
      >
        <img
          alt="Avatar md"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="48"
          sizes="(max-width: 768px) 100vw, 33vw"
          src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=3840&q=75"
          srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=3840&q=75 3840w"
          style="color: transparent;"
          width="48"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="inline-flex rounded-full overflow-hidden shadow-md"
    >
      <img
        alt="Avatar md"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="48"
        sizes="(max-width: 768px) 100vw, 33vw"
        src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=3840&q=75"
        srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-md.webp&w=3840&q=75 3840w"
        style="color: transparent;"
        width="48"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
